server/message_queue_service.py

```diff
@@ -31,12 +33,14 @@ def __init__(self) -> None:
         config.register_callback("MQ_PORT", self.reconnect)
 
     async def initialize(self) -> None:
-        if self._connection is not None:
-            return
+        async with self._initialization_lock:
+            if self._connection is not None:
+                return
 
-        await self._connect()
+            self._is_ready = await self._connect()
```

Comment:
why not make `_connect` set the attribute directly?

Reply:
Because I also want to call `_connect` from `reconnect` and there I don't want to set the flag until after the previous exchanges have been re-declared.

Reply:
Hmm, maybe it would be better to propagate exceptions then. I feel like passing around a "
success" flag as return value is pretty clunky. Gives me nightmares about C.

Reply:
Sure thing.